### PR TITLE
Remove volume - add words to docs to make using external volume more obvious

### PR DESCRIPTION
**Pull request by dceejay**
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Proposed changes

Add `-v node_red_data:/data` to most docker run command line examples so that users start to use external bind mounts for persisting data by default.

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md)
- x ] For non-bugfix PRs, I have discussed this change on the mailing list/slack team.
- [ ] I have run `grunt` to verify the unit tests pass
- [ ] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,13 +15,14 @@ Previous 0.20.x versions are still available at https://hub.docker.com/r/nodered
...
 
-        docker run -it -p 1880:1880 --name mynodered nodered/node-red
+        docker run -it -p 1880:1880 -v node_red_data:/data --name mynodered nodered/node-red
 
 Let's dissect that command:
 
         docker run              - run this container, initially building locally if necessary
         -it                     - attach a terminal session so we can see what is going on
         -p 1880:1880            - connect local port 1880 to the exposed internal port 1880
+        -v node_red_data:/data - mount the host node_red_data directory to the container /data directory so any changes made to flows are persisted
```
Comment (RaymondMouthaan) [inline README.md:25]: space missing

File: .docker/Dockerfile.alpine [hunk 2]
```diff
@@ -88,7 +88,7 @@ ENV NODE_RED_VERSION=$NODE_RED_VERSION \
 # ENV NODE_RED_ENABLE_SAFE_MODE=true    # Uncomment to enable safe start mode (flows not running)
 # ENV NODE_RED_ENABLE_PROJECTS=true     # Uncomment to enable projects option
 
-# User configuration directory volume
+# User configuration directory volume (if required)
```
Comment (RaymondMouthaan) [inline .docker/Dockerfile.alpine:91]: would recommend just to remove it instead of putting it into comment

File: README.md [hunk 3]
```diff
@@ -439,6 +440,8 @@ docker id number and be running on a random port... to find out run
 You can now point a browser to the host machine on the tcp port reported back, so in the example
 above browse to  `http://{host ip}:49154`
 
+**NOTE**: as this does not mount the `/data` volume externally any changes to flows will not be saved and if the container is debuilt or upgraded these will be lost. The volume may persist on the host filing sysem and can probably be retrieved and remounted if required.
```
Comment (RaymondMouthaan) [inline README.md:443]: debuilt = redeployed

File: docker-custom/Dockerfile.custom [hunk 4]
```diff
@@ -81,7 +81,7 @@ ENV NODE_RED_VERSION=$NODE_RED_VERSION \
 # ENV NODE_RED_ENABLE_SAFE_MODE=true    # Uncomment to enable safe start mode (flows not running)
 # ENV NODE_RED_ENABLE_PROJECTS=true     # Uncomment to enable projects option
 
-# User configuration directory volume
+# User configuration directory volume (if required)
```
Comment (RaymondMouthaan) [inline docker-custom/Dockerfile.custom:84]: reasonable to leave the section here as this is an example Dockerfile 👍

File: docker-custom/Dockerfile.debian [hunk 5]
```diff
@@ -82,7 +82,7 @@ ENV NODE_RED_VERSION=$NODE_RED_VERSION \
 # ENV NODE_RED_ENABLE_SAFE_MODE=true    # Uncomment to enable safe start mode (flows not running)
```
Comment (RaymondMouthaan) [inline docker-custom/Dockerfile.debian:82]: nice to have a custom for Debian as well 👍

**Review by RaymondMouthaan — APPROVED** [review]
add a few nits to the pr

File: .docker/Dockerfile.alpine (hunk 2 above)
Comment (dceejay) [reply .docker/Dockerfile.alpine]: ok

File: README.md (hunk 1 above)
Comment (dceejay) [reply README.md]: yes

File: README.md (hunk 3 above)
Comment (dceejay) [reply README.md:443]: fixed

File: docker-custom/Dockerfile.custom (hunk 4 above)
Comment (dceejay) [reply docker-custom/Dockerfile.custom]: well - let's just remove them all.

**Review by RaymondMouthaan — APPROVED** [review]

**Review by knolleary — APPROVED** [review]

**Comment by knolleary** [comment]
We need to remember the feedback from slack that some container platforms (eg OpenShift) will only let you mount volumes that are declared in the docker file. We'll need to address that (such as creating an alternative image) before we release this generally.

**Comment by dceejay** [comment]
Slack link referred to is here - https://node-red.slack.com/archives/C1YH60001/p1599905020140400

**Pull request merged**